api/src/main/java/com/epam/pipeline/manager/cloud/gcp/GCPVMService.java

```diff
@@ -82,22 +87,40 @@ public Instance getRunningInstanceByRunId(final GCPRegion region, final String r
         }
     }
 
-    public Optional<InstanceTerminationState> getTerminationState(final GCPRegion region,
-                                                                  final String instanceId) {
+    public Optional<InstanceTerminationState> getTerminationState(final GCPRegion region, final String instanceId) {
+        Optional<InstanceTerminationState> result = Optional.empty();
```

Comment:
I'd suggest not to introduce a mutable variable. And just return result instead of assigning a new value.